image/cache/cache.go

```diff
@@ -132,28 +137,31 @@ func (ic *ImageCache) restoreCachedImage(parent, target *image.Image, cfg *conta
 		return "", errors.Wrap(err, "failed to marshal image config")
 	}
 
-	imgID, err := ic.store.Create(config)
+	imgID, err := ic.store.Create(ctx, config)
 	if err != nil {
 		return "", errors.Wrap(err, "failed to create cache image")
 	}
 
 	if parent != nil {
-		if err := ic.store.SetParent(imgID, parent.ID()); err != nil {
+		if err := ic.store.SetParent(ctx, imgID, parent.ID()); err != nil {
 			return "", errors.Wrapf(err, "failed to set parent for %v to %v", target.ID(), parent.ID())
 		}
 	}
 	return imgID, nil
 }
 
-func (ic *ImageCache) isParent(imgID, parentID image.ID) bool {
-	nextParent, err := ic.store.GetParent(imgID)
+func (ic *ImageCache) isParent(ctx context.Context, imgID, parentID image.ID) (bool, error) {
+	nextParent, err := ic.store.GetParent(ctx, imgID)
 	if err != nil {
-		return false
+		if errdefs.IsNotFound(err) {
```

Comment:
Could use some eyes on this one as well /cc @tonistiigi - I see this was implemented in https://github.com/moby/moby/commit/690882c2e79c3f3742c709cf158584e61594ba00, which ignored the errors (and returned `false` for error conditions); perhaps there was a reason for this.